serverside_challenge_1/challenges/toru_shimizu/spec/requests/prices_spec.rb

```diff
@@ -0,0 +1,155 @@
+  service = PricesService.new
+
+  describe 'GET /api/prices' do
+    context 'リクエストパラメータが正常な場合' do
+      it '200 OK が返ってくる' do
+        get api_prices_path, params: { ampere: 10, volume: 300 }
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'プランごとの電気料金を取得できる' do
+        get api_prices_path, params: { ampere: 30, volume: 300 }
+
+        expected = service.create_response(30, 300, data['companies'])
+        expect(body).to eq(expected)
+      end
+    end
+
+    context 'リクエストパラメータが不正な場合' do
+      context 'ampere' do
+        context '存在しない場合' do
+          it '400 Bad Request が返ってくる' do
+            get api_prices_path, params: { volume: 100 }
+            expect(response).to have_http_status(400)
+          end
+
+          it 'アンペアは必須です とメッセージが返ってくる' do
+            get api_prices_path, params: { volume: 100 }
+
+            expect(body.length).to eq(1)
+            expect(body[0]['message']).to eq('アンペアは必須です')
+          end
+        end
+
+        context '契約可能な範囲外の場合' do
+          it '400 Bad Request が返ってくる' do
+            get api_prices_path, params: { ampere: 100, volume: 100 }
+            expect(response).to have_http_status(400)
+          end
+
+          it '契約対象外のアンペアです とメッセージが返ってくる' do
+            get api_prices_path, params: { ampere: 100, volume: 100 }
+
+            expect(body.length).to eq(1)
+            expect(body[0]['message']).to eq('契約対象外のアンペアです')
+          end
+        end
+      end
+
+      context 'volume' do
+        context '存在しない場合' do
+          it '400 Bad Request が返ってくる' do
+            get api_prices_path, params: { ampere: 100 }
+            expect(response).to have_http_status(400)
+          end
+
+          it '使用量は必須ですと メッセージが返ってくる' do
+            get api_prices_path, params: { ampere: 10 }
+
+            expect(body.length).to eq(1)
+            expect(body[0]['message']).to eq('使用量は必須です')
+          end
+        end
+
+        context '使用量が不正な値の場合' do
+          context '負の数' do
+            it '400 Bad Request が返ってくる' do
+              get api_prices_path, params: { ampere: 10, volume: -100 }
+
+              expect(response).to have_http_status(400)
+            end
+
+            it '使用量は0以上の値を入力してください とメッセージが返ってくる' do
+              get api_prices_path, params: { ampere: 10, volume: -100 }
+
+              expect(body.length).to eq(1)
+              expect(body[0]['message']).to eq('使用量は0以上の値を入力してください')
+            end
+          end
+
+          context '整数でない場合' do
+            it '400 Bad Request が返ってくる' do
+              get api_prices_path, params: { ampere: 10, volume: 10.10 }
+
+              expect(response).to have_http_status(400)
+            end
+
+            it '使用量は整数で入力してください とメッセージが返ってくる' do
+              get api_prices_path, params: { ampere: 10, volume: 10.10 }
+
+              expect(body.length).to eq(1)
+              expect(body[0]['message']).to eq('使用量は整数で入力してください')
+            end
+          end
```

Comment:
テストの観点を結果で分けて2つのテストにしているのだと思いますが、Inputの種類で分けてこれらのテストは1つのit ~ endでまとめて良いと思います。実行スピードも早くなりますし、脳にも優しいと思います。

```suggestion
          context '整数でない場合' do
            it '400 Bad Request とエラーメッセージが返却される' do
              get api_prices_path, params: { ampere: 10, volume: 10.10 }

              expect(response).to have_http_status(400)
              expect(body[0]['message']).to eq('使用量は整数で入力してください')
            end
          end
```